util/src/main/java/io/grpc/util/AdvancedTlsX509KeyManager.java

```diff
@@ -131,14 +131,17 @@ public Closeable updateIdentityCredentialsFromFile(File keyFile, File certFile,
       throw new GeneralSecurityException(
           "Files were unmodified before their initial update. Probably a bug.");
     }
+    if (checkNotNull(unit, "unit").toMinutes(period) < MINIMUM_REFRESH_PERIOD) {
+      log.log(Level.INFO,
+          "Provided refresh period of {0} {1} is too small. Default value of {2} minute(s) "
+          + "will be used.", new Object[] {period, unit.name(), MINIMUM_REFRESH_PERIOD});
+      period = MINIMUM_REFRESH_PERIOD;
+      unit = TimeUnit.MINUTES;
+    }
     final ScheduledFuture<?> future =
-        executor.scheduleWithFixedDelay(
+        checkNotNull(executor, "executor").scheduleWithFixedDelay(
```

Comment:
What's the goal with this checkNotNull being added?

Reply:
I saw a 'checkNotNull' approach in this class for one public API - 
https://github.com/grpc/grpc-java/blob/8aaace12eb8697a2dd841d2261b0051a1414f0c8/util/src/main/java/io/grpc/util/AdvancedTlsX509KeyManager.java#L111

and just added it for other public API methods. In this particular case without it we'd get a raw NPE at 
https://github.com/grpc/grpc-java/blob/8aaace12eb8697a2dd841d2261b0051a1414f0c8/util/src/main/java/io/grpc/util/AdvancedTlsX509KeyManager.java#L135
if the user passes null for 'executor' param.